test/SecTester.Scan.Tests/ScanTests.cs

```diff
@@ -1,53 +1,228 @@
+using Microsoft.Extensions.Logging;
+using NSubstitute.ClearExtensions;
+using SecTester.Scan.Tests.Fixtures;
+
 namespace SecTester.Scan.Tests;
 
-public class ScanTests
+public class ScanTests : ScanFixture
 {
+  private readonly Scans _scans = Substitute.For<Scans>();
+  private readonly LoggerMock _logger = Substitute.For<LoggerMock>();
+
+  public override void Dispose()
+  {
+    _scans.ClearSubstitute();
+    _logger.ClearSubstitute();
+
+    base.Dispose();
+  }
+
+  [Fact]
+  public void Constructor_GivenNullId_ThrowError()
+  {
+    // act
+    Action act = () => new Scan(null!, _scans, _logger, new ScanOptions());
+
+    // assert
+    act.Should().Throw<ArgumentNullException>().WithMessage("*id*");
+  }
+
   [Fact]
   public void Constructor_GivenNullScans_ThrowError()
   {
     // act
-    var act = () => new Scan(null!, new ScanOptions());
+    Action act = () => new Scan(ScanId, null!, _logger, new ScanOptions());
 
     // assert
     act.Should().Throw<ArgumentNullException>().WithMessage("*scans*");
   }
 
+  [Fact]
+  public void Constructor_GivenNullLogger_ThrowError()
+  {
+    // act
+    Action act = () => new Scan(ScanId, _scans, null!, new ScanOptions());
+
+    // assert
+    act.Should().Throw<ArgumentNullException>().WithMessage("*logger*");
+  }
+
   [Fact]
   public void Constructor_GivenNullScanOptions_ThrowError()
+  {
+    // act
+    Action act = () => new Scan(ScanId, _scans, _logger, null!);
+
+    // assert
+    act.Should().Throw<ArgumentNullException>().WithMessage("*options*");
+  }
+
+  [Fact]
+  public async Task Active_DefaultValue_ReturnsTrue()
+  {
+    // act
+    await using var scan = new Scan(ScanId, _scans, _logger, new ScanOptions());
+
+    // assert
+    scan.Active.Should().BeTrue();
+  }
+
+  [Fact]
+  public async Task Done_DefaultValue_ReturnsTrue()
+  {
+    // act
+    await using var scan = new Scan(ScanId, _scans, _logger, new ScanOptions());
+
+    // assert
+    scan.Done.Should().BeFalse();
+  }
+
+  [Fact]
+  public async Task Status_ReturnsState()
```

Comment:
I don't see any test to verify that `Status` return the last obtained status when the status is done